### PR TITLE
test: add integration tests with enzyme, and use jest instead of mocha

### PR DESCRIPTION
**Pull request by frederickfogerty**
## Description

Previously this project used `skin-deep` to test this component using shallow rendering. Although this was useful for a lot of testing scenarios, it doesn't allow integration tests (i.e. interacting with an actual DOM) to be created. Using a testing library which allows DOM integration tests permits a more comprehensive test suite to be created. Moving to using enzyme allows this.

### New Feature

- [x] Add some [steps](#steps-to-test) so we can test your cool new feature!
- [x] The PR title is in the conventional commit format: e.g. `feat(<area>): added new way to do this cool thing #issue-number`

## Steps to Test

Steps:

1.  Clone this PR and run `npm install`
2.  Run `npm run test` and observe that all the tests now use Jest

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,8 +18,8 @@
     "prepublish": "npm run build",
     "pretty": "prettier --write \"src/**/*.js\"",
     "release": "standard-version",
-    "test:watch": "cross-env BABEL_ENV=commonjs mocha --require babel-core/register --recursive src/*.test.js --watch --reporter min",
-    "test": "npm run pretty && cross-env BABEL_ENV=commonjs mocha --require babel-core/register --recursive src/*.test.js"
+    "test:watch": "jest --watch",
+    "test": "jest"
```
Comment (jayeb) [inline package.json]: Well, this sure is nicer.

**Comment by frederickfogerty** [comment]
It's about 50% a file change. There are significant changes to the test files (rewrite to enzyme, renaming, the addition of integration tests). I didn't know about git mv - I'll try to rebase these commits to clean up the diff.

**Comment by frederickfogerty** [comment]
@jayeb I've gone through and rebased this entire PR off the master locally, using `git mv` where applicable, and pushed it to a new branch. Unfortunately, the diff is the same. For review purposes, the important changes are in `test/integration/react-imgix.test.js` and `test/unit/react-imgix.test.js`. 

To preserve the line versioning I think the best way is to, once this PR has been approved, rebase the commits to be as few as possible, and then do a normal merge commit (i.e. not squash and merge) which *should* preserve the changes.

File: test/unit/react-imgix.test.js [hunk 2]
```diff
@@ -0,0 +1,606 @@
...
+  it("the rendered image's src should be set", () => {
+    expect(
+      renderImage()
+        .find("img")
+        .props().src
+    ).toContain(src);
+  });
+});
+
+describe("When in image mode", () =>
```
Comment (jayeb) [inline test/unit/react-imgix.test.js]: For the sake of consistency, can we wrap this block in curlies?

**Pull request merged**